src/rgw/rgw_bucket.cc

```diff
@@ -2995,10 +2995,12 @@ int RGWMetadataHandlerPut_BucketInstance::put_check()
     bci.info.bucket.name = bucket_name;
     bci.info.bucket.bucket_id = bucket_instance;
     bci.info.bucket.tenant = tenant_name;
-    ret = bihandler->svc.zone->select_bucket_location_by_rule(bci.info.placement_rule, &rule_info);
-    if (ret < 0) {
-      ldout(cct, 0) << "ERROR: select_bucket_placement() returned " << ret << dendl;
-      return ret;
+    if (bihandler->svc.zone->zone_is_writeable()) {
```

Comment:
`RGWSI_Zone::zone_is_writeable()` is a combination of `RGWSyncModuleRef::supports_writes()` and `!RGWZone::is_read_only()`

this condition should only look at `RGWSyncModuleRef::supports_writes()`, though, because `RGWZone::is_read_only()` could be changed later with `zone modify --read-only=false`. then we'd be left with bucket metadata whose placement doesn't match its peer zones, and we would actually be writing to its placement pools

can we expose a separate function for this, ie:
```c++
bool RGWSI_Zone::sync_module_supports_writes() const { return writeable_zone; }
```

also let's add a comment for this change:
```c++
// if the sync module never writes data, don't require the zone to specify all placement targets
```

Reply:
@cbodley Thanks a lot. I have modified it as you suggested.